lms/services/blackboard_api/_schemas.py

```diff
@@ -8,46 +8,35 @@
 from lms.validation._base import RequestsResponseSchema
 
 
-class _FileSchema(Schema):
-    """Shared base schema for Blackboard file dicts."""
-
-    class Meta:
-        unknown = EXCLUDE
-
-    id = fields.Str(required=True)
-    display_name = fields.Str(data_key="name", required=True)
-    updated_at = fields.Str(data_key="modified", required=True)
-    mime_type = fields.Str(data_key="mimeType")
+class BlackboardListFilesSchema(RequestsResponseSchema):
+    """Schema for Blackboard API /courses/{courseId}/resources responses."""
 
-    @post_load
-    def post_load(self, data, **_kwargs):  # pylint:disable=no-self-use
-        data["id"] = f"blackboard://content-resource/{data['id']}/"
-        return data
+    class FileSchema(Schema):
+        """Schema for individual Blackboard file dicts."""
 
+        class Meta:
+            unknown = EXCLUDE
 
-class BlackboardListFilesSchema(RequestsResponseSchema):
-    """Schema for Blackboard API /courses/{courseId}/resources responses."""
+        id = fields.Str(required=True)
+        name = fields.Str(required=True)
+        modified = fields.Str(required=True)
+        mimeType = fields.Str()
 
-    results = fields.List(fields.Nested(_FileSchema), required=True)
+    results = fields.List(fields.Nested(FileSchema), required=True)
 
     @post_load
     def post_load(self, data, **_kwargs):  # pylint:disable=no-self-use
-        pdf_files = []
-
-        for file in data["results"]:
-            if file.get("mime_type") == "application/pdf":
-                # Delete mime_type: we don't want to send it to the frontend.
-                del file["mime_type"]
-                pdf_files.append(file)
-
-        return pdf_files
+        return data["results"]
```

Comment:
It turned out to make everything simpler if the schema _does_ still pop out the results key